modules/engine/src/testing/listeners.spec.ts

```diff
@@ -741,4 +745,127 @@ describe(testName, () => {
+        const test = createTestChannelState("create", { networkContext: { chainId } });
+        const meta = {
+          routingId: mkHash("0xaaa"),
+          path: [
+            {
+              recipient: signer.publicIdentifier,
+              recipientAssetId: AddressZero,
+              recipientChainId: chainId,
+            },
+          ],
+        } as RouterSchemas.RouterMeta;
+        test.channel.latestUpdate.details.meta = meta;
+        test.transfer.responderIdentifier = signer.publicIdentifier;
+        test.transfer.meta = meta;
+        const event: ChannelUpdateEvent = {
+          updatedChannelState: test.channel,
+          updatedTransfer: test.transfer,
+        };
+        const evts = getEngineEvtContainer();
+
+        evts.CONDITIONAL_TRANSFER_ROUTING_COMPLETE.attachOnce((data) => {
+          expect(data).to.deep.eq({
+            initiatorIdentifier: test.transfer.initiatorIdentifier,
+            responderIdentifier: test.transfer.responderIdentifier,
+            routingId: test.transfer.meta.routingId,
+            meta: test.transfer.meta,
+          } as ConditionalTransferRoutingCompletePayload);
+          res();
+        });
+        handleConditionalTransferCreation(
+          event,
+          store,
+          chainService as any,
+          chainAddresses,
+          evts,
+          log,
+          signer,
+          messaging,
+        ).catch(rej);
+      });
+    });
+
+    it("should pass message back to sender as the exit router", async () => {
+      const signer = getRandomChannelSigner();
+      const test = createTestChannelState("create", { networkContext: { chainId } });
+      const meta = {
+        routingId: mkHash("0xaaa"),
+        path: [
+          {
+            recipient: bob.publicIdentifier,
+            recipientAssetId: AddressZero,
+            recipientChainId: chainId,
+          },
+        ],
+      } as RouterSchemas.RouterMeta;
+      test.channel.latestUpdate.details.meta = meta;
+      test.transfer.initiatorIdentifier = signer.publicIdentifier;
+      test.transfer.responderIdentifier = bob.publicIdentifier;
+      test.transfer.meta = meta;
+
+      const event: ChannelUpdateEvent = {
+        updatedChannelState: test.channel,
+        updatedTransfer: test.transfer,
+      };
+      const evts = getEngineEvtContainer();
+
+      // resolve to
+      store.getTransfersByRoutingId.resolves([
+        test.transfer,
+        { ...test.transfer, initiatorIdentifier: alice.publicIdentifier, responderIdentifier: signer.publicIdentifier },
+      ]);
+
+      evts.CONDITIONAL_TRANSFER_ROUTING_COMPLETE.attachOnce((data) => {
+        expect(data).to.deep.eq({
+          initiatorIdentifier: test.transfer.initiatorIdentifier,
+          responderIdentifier: test.transfer.responderIdentifier,
+          routingId: test.transfer.meta.routingId,
+          meta: test.transfer.meta,
+        } as ConditionalTransferRoutingCompletePayload);
+      });
```

Comment:
How can you ensure the event was emitted properly and the structure is asserted if this isn't a `waitFor`? It could be that the other expects succeed, and the test suite is terminated.